src/main/java/services/notification_sending/EmailNotificationPresenter.java

```diff
@@ -0,0 +1,81 @@
+package services.notification_sending;
+
+import java.util.Properties;
+import javax.mail.*;
+import javax.mail.internet.*;
+
+/**
+ * Sends out email notifications.
+ * Keeps a record of whether the user enabled the email notification settings.
+ */
+public class EmailNotificationPresenter implements NotificationPresenter, SettingsRegistry {
+    private boolean enabled;
+    private String senderEmail;
+    private String password;
+    private String userEmail;
+    private String subject;
+
+    EmailNotificationPresenter(boolean enabled, String senderEmail, String password, String userEmail, String subject) {
+        this.enabled = enabled;
+        this.senderEmail = senderEmail;
+        this.password = password;
+        this.userEmail = userEmail;
+        this.subject = subject;
+    }
+
+    @Override
+    public void presentNotification(String message) {
+        //Get properties object
+        Properties props = new Properties();
+        props.put("mail.smtp.host", "smtp.gmail.com");
+        props.put("mail.smtp.socketFactory.port", "465");
+        props.put("mail.smtp.socketFactory.class",
+                "javax.net.ssl.SSLSocketFactory");
+        props.put("mail.smtp.auth", "true");
+        props.put("mail.smtp.port", "465");
+
+        //get Session
+        Session session = Session.getDefaultInstance(props,
+                new javax.mail.Authenticator() {
+                    protected PasswordAuthentication getPasswordAuthentication() {
+                        return new PasswordAuthentication(senderEmail, password);
+                    }
+                });
+
+        //compose message
+        try {
+            MimeMessage msg = new MimeMessage(session);
+            msg.addRecipient(Message.RecipientType.TO,new InternetAddress(userEmail));
+            msg.setSubject(subject);
+            msg.setText(message);
+            //send message
+            Transport.send(msg);
+            System.out.println("message sent successfully");
+        } catch (MessagingException e) {throw new RuntimeException(e);}
```

Comment:
Which part of the composing or sending actually throws this error? This try block is encompassing a number of 3rd party code so it's difficult to tell what you're actually expecting to be thrown.

Also I'm thinking that we either want to silence this exception or throw one of our own because otherwise this'll stop every other notification system from working. If we make our own exception then whatever class sends these notifs can just catch the error with just one exception (OCP), but I feel like in the end it'll just silence the exception there anyway.

Reply:
Sorry I'm a bit lazy so I will just show the screenshoots. Based on these documentations, I don't think we are able to check and deal with some exceptions by ourselves.
<img width="447" alt="截屏2021-12-04 上午10 23 05" src="https://user-images.githubusercontent.com/71200653/144715058-cd94bfa0-afcf-4b55-aca3-c7190bb006c2.png">
<img width="546" alt="截屏2021-12-04 上午10 23 16" src="https://user-images.githubusercontent.com/71200653/144715062-93201fba-2625-4036-8c68-6ec2872cb96d.png">
<img width="529" alt="截屏2021-12-04 上午10 23 22" src="https://user-images.githubusercontent.com/71200653/144715066-3e7b62ff-948a-4db6-8b35-a9d293f3e96b.png">